flink-runtime/src/main/scala/org/apache/flink/runtime/taskmanager/TaskManager.scala

```diff
@@ -547,7 +552,7 @@ class TaskManager(
 
         // successful registration. associate with the JobManager
         // we disambiguate duplicate or erroneous messages, to simplify debugging
-        case AcknowledgeRegistration(_, leaderSessionID, jobManager, id, blobPort) =>
+        case AcknowledgeRegistration(_, leaderSessionID, jobManager, id, blobPort, jobManagerID) =>
```

Comment:
The initial AcknowledgeRegistration message can get lost and then the JM will send a `AlreadyRegistered` message.